src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php

```diff
@@ -39,46 +39,39 @@ protected function setUp(): void
         $this->encoder->setSerializer($serializer);
     }
 
-    public function testEncodeScalar()
+    /**
+     * @dataProvider validEncodeProvider
+     */
+    public function testEncode(string $expected, mixed $data, array $context = [])
     {
-        $obj = new ScalarDummy();
-        $obj->xmlFoo = 'foo';
-
-        $expected = '<?xml version="1.0"?>'."\n".
-            '<response>foo</response>'."\n";
-
-        $this->assertEquals($expected, $this->encoder->encode($obj, 'xml'));
+        static::assertSame($expected, $this->encoder->encode($data, 'xml', $context));
```

Comment:
Why this change? We don't call the assertion methods in a static manner anywhere else in the tests, do we?

Reply:
Damn, I replaced that already, as Fabien also added that feedback

Reply:
Fixed